pkg/scheduler/core/generic_scheduler.go

```diff
@@ -903,11 +904,44 @@ func pickOneNodeForPreemption(nodesToVictims map[*v1.Node]*schedulerapi.Victims)
 			lenNodes2++
 		}
 	}
+	if lenNodes2 == 1 {
+		return minNodes2[0]
+	}
+
+	// There are a few nodes with same number of pods.
+	// Find the node that satisfies latest(earliestStartTime(all highest-priority pods on node))
+	latestStartTime := util.GetEarliestPodStartTime(nodesToVictims[minNodes2[0]])
+	if latestStartTime == nil {
+		// If the earliest start time of all pods on the 1st node is nil, just return it,
+		// which is not expected to happen.
```

Comment:
Should we check the second node's start time (in this case) ?

Reply:
Later on line 925:
```
                        klog.Errorf("earliestStartTime is nil for node %s. Should not reach here.", node)
                        continue
```
I think using a loop for finding start time around line 913 would make the code more consistent.

Reply:
@tedyu `latestStartTime` should never be nil. If for some unexpected reason it becomes nil, we prefer to see an error message and investigate the issue instead of tolerating the error.

Reply:
In that case, panic'ing would be proper choice in this scenario.

Reply:
> In that case, panic'ing would be proper choice in this scenario.

Some test cases will panic because they perhaps do not set `StartTime` :(

Reply:
Interesting.
In another popular Apache project, my test encounters NPE due to null being passed by test(s) to configure() (in production, it wouldn't be null).
I am modifying the test not to do that.

For k8s, we should do the same - tests should mimic what happens in production.